File: Demo Objective-C/LGSideMenuControllerDemo/LeftViewCell.m
//
//  LeftViewCell.m
//  LGSideMenuControllerDemo
//
//  Created by Grigory Lutkov on 26.04.15.
//  Copyright © 2015 Grigory Lutkov <Friend.LGA@gmail.com>. All rights reserved.
//

#import "LeftViewCell.h"

@implementation LeftViewCell

- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier {
    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
    if (self) {
        self.selectionStyle = UITableViewCellSelectionStyleNone;

        self.backgroundColor = [UIColor clearColor];

        self.textLabel.font = [UIFont boldSystemFontOfSize:16.0];

        // -----

        self.separatorView = [UIView new];
        [self addSubview:self.separatorView];
    }
    return self;
}

- (void)layoutSubviews {
    [super layoutSubviews];

    self.textLabel.textColor = self.tintColor;
    self.separatorView.backgroundColor = [self.tintColor colorWithAlphaComponent:0.4];

    CGFloat height = [UIScreen mainScreen].scale == 1.0 ? 1.0 : 0.5;

    self.separatorView.frame = CGRectMake(0.0, self.frame.size.height-height, self.frame.size.width*0.9, height);
}

- (void)setHighlighted:(BOOL)highlighted animated:(BOOL)animated {
    self.textLabel.textColor = highlighted ? [UIColor colorWithRed:0.0 green:0.5 blue:1.0 alpha:1.0] : self.tintColor;
}

@end

File: Demo/_shared_files/SideMenuController/LeftViewController/LeftViewCell.swift
//
//  LeftViewCell.swift
//  LGSideMenuControllerDemo
//

import Foundation
import UIKit

private let textFont = UIFont.boldSystemFont(ofSize: 16.0)
private let fillColorNormal = UIColor(white: 0.0, alpha: 0.2)
private let fillColorNormalInverted = UIColor(white: 1.0, alpha: 0.2)
private let fillColorHighlighted = UIColor(white: 1.0, alpha: 0.6)
private let textColorNormal: UIColor = .white
private let textColorHighlighted: UIColor = .black

class LeftViewCell: UITableViewCell {

    var isFirst: Bool = false {
        didSet {
            guard let backgroundView = self.backgroundView as? LeftViewCellBackgroundView else { return }
            backgroundView.isFirstCell = isFirst
        }
    }

    var isLast: Bool = false {
        didSet {
            guard let backgroundView = self.backgroundView as? LeftViewCellBackgroundView else { return }
            backgroundView.isLastCell = isLast
        }
    }

    var isFillColorInverted: Bool = false

    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)

        selectionStyle = .none
        backgroundColor = .clear

        backgroundView = LeftViewCellBackgroundView()
        setBackgroundColor(getFillColorNormal())

        textLabel!.font = textFont
        textLabel!.textColor = textColorNormal
        textLabel!.numberOfLines = 1
        textLabel!.textAlignment = .left
    }

    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    override func layoutSubviews() {
        super.layoutSubviews()

        backgroundView!.frame = CGRect(x: 0.0,
                                       y: 0.0,
                                       width: bounds.width - 16.0,
                                       height: bounds.height)

        textLabel!.frame = CGRect(x: 8.0,
                                  y: 0.0,
                                  width: bounds.width - 16.0 - 16.0,
                                  height: bounds.height)
    }

    override func setHighlighted(_ highlighted: Bool, animated: Bool) {
        super.setHighlighted(highlighted, animated: animated)
        setBackgroundColor(highlighted ? fillColorHighlighted : getFillColorNormal())
        textLabel!.textColor = highlighted ? textColorHighlighted : textColorNormal
    }

    func getFillColorNormal() -> UIColor {
        return isFillColorInverted ? fillColorNormalInverted : fillColorNormal
    }

    func setBackgroundColor(_ color: UIColor) {
        let backgroundView = self.backgroundView as! LeftViewCellBackgroundView
        backgroundView.fillColor = color
    }

}
